packer/linux/conf/buildkite-agent/scripts/terminate-instance

```diff
@@ -2,14 +2,51 @@
 
 set -euo pipefail
 
-echo "sleeping for 10 seconds before terminating instance to allow agent logs to drain to cloudwatch..."
+terminate() {
+  aws autoscaling terminate-instance-in-auto-scaling-group \
+    --region "$1" \
+    --instance-id "$2" \
+    --should-decrement-desired-capacity
+}
+
+mark_as_unhealthy() {
+  aws autoscaling set-instance-health \
+    --region "$1" \
+    --instance-id "$2" \
+    --health-status Unhealthy
```

Comment:
The ps1 includes `--no-should-respect-grace-period` - I think this should have it too, if the goal is to emulate `shutdown` (and where someone has configured a grace period).

Reply:
Looks like I copy-pasted that from your PR 😅.

The grace period is set by the [`HealthCheckGracePeriod`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-autoscaling-autoscalinggroup.html#cfn-autoscaling-autoscalinggroup-healthcheckgraceperiod) field of an `AWS::AutoScaling::AutoScalingGroup`, but that is not set here: https://github.com/buildkite/elastic-ci-stack-for-aws/blob/5b02d9550e3f8274028edecbfe504269718912ae/templates/aws-stack.yml#L1253-L1304

So the default of `0s` is what all stacks have by default. Given this, I think if the user has gone out of their way to set a grace period, we should not interfere.

Reply:
I'm split on this. On the one hand, that's a fairly good argument for respecting the grace period. But on the other, why would they configure `BuildkiteTerminateInstanceAfterJob` at all (if they also want a grace period, in which to... recover the agent?)

Reply:
IDK, but if explicitly did both, they may have thought of a reason we haven't.